tests/test_agents.py

```diff
@@ -644,29 +637,29 @@ def get_weather(location: str, celsius: bool = False) -> str:
             device_map="auto",
             do_sample=False,
         )
-        agent = ToolCallingAgent(model=model, tools=[get_weather], max_steps=1)
+        agent = ToolCallingAgent(model=model, tools=[weather_api], max_steps=1)
         agent.run("What's the weather in Paris?")
         assert agent.memory.steps[0].task == "What's the weather in Paris?"
-        assert agent.memory.steps[1].tool_calls[0].name == "get_weather"
+        assert agent.memory.steps[1].tool_calls[0].name == "weather_api"
         step_memory_dict = agent.memory.get_succinct_steps()[1]
-        assert step_memory_dict["model_output_message"].tool_calls[0].function.name == "get_weather"
+        assert step_memory_dict["model_output_message"].tool_calls[0].function.name == "weather_api"
         assert step_memory_dict["model_output_message"].raw["completion_kwargs"]["max_new_tokens"] == 100
         assert "model_input_messages" in agent.memory.get_full_steps()[1]
 
 
 class TestMultiStepAgent:
     def test_step_number(self):
         fake_model = MagicMock()
-        agent = MultiStepAgent(tools=[], model=fake_model)
+        agent = MultiStepAgent(tools=[], model=fake_model, max_steps=2)
         assert hasattr(agent, "step_number"), "step_number attribute should be defined"
         assert agent.step_number == 0, "step_number should be initialized to 0"
-        agent.run("Test task", single_step=True)
-        assert hasattr(agent, "step_number"), "step_number attribute should be defined"
+        agent.run("Test task")
+        assert hasattr(agent, "step_number", 3), "step_number attribute should be defined"
```

Comment:
I think the signature is wrong.
```suggestion
        assert hasattr(agent, "step_number"), "step_number attribute should be defined"
```